### ISSUE TITLE
[Bug]: 'Rescan subscriptions' button doesn't detect new videos for playlists

### ISSUE THREAD

**Issue by trinhit** [tubearchivist/tubearchivist#686]
### I've read the documentation

- [X] I'm running the latest version of Tube Archivist and have read the [release notes](https://github.com/tubearchivist/tubearchivist/releases/latest).
- [X] I have read the [how to open an issue](https://github.com/tubearchivist/tubearchivist/blob/master/CONTRIBUTING.md#how-to-open-an-issue) guide, particularly the [bug report](https://github.com/tubearchivist/tubearchivist/blob/master/CONTRIBUTING.md#bug-report) section.

### Operating System

Synology

## Describe the bug
I subscribed to a playlist. When clicking on the **Rescan subscriptions** button for the first time, all videos in that playlist were added into download queue. After some time, more videos were added into the same playlist. However, clicking on **Rescan subscriptions** button again didn't detect newly added videos.

## Steps To Reproduce
Pre-condition: I tested with a clean installation of TA with all default settings.

Step 1: Go to YouTube, create a public or unlisted playlist. Add a few videos into the playlist. Let's say the playlist has 3 videos at this moment.

Step 2: On TubeArchivist, go to `/playlist` page. Input the URL/ID of the playlist on Step 1 and click on **Subscribe** button.

Step 2: Go to `/downloads` page. Click on **Rescan subscriptions** button. We'll see 3 videos in the download queue.

Step 3: Go back to YouTube, add a few more videos into the same playlist.

Step 4: On TubeArchivist, go to `/downloads` page. Click on **Rescan subscriptions** button. Observation: The newly added videos on Step 3 don't show up on the download queue.

## Expected behavior
Newly added videos of subscribed playlists should be added into download queue when clicking on **Rescan subscriptions** button.

### Relevant log output

```shell
[2024-03-17 17:26:06,386: INFO/MainProcess] Task update_subscribed[ab98fc35-31f5-4070-93a6-9054022559a2] received
[2024-03-17 17:26:06,387: WARNING/ForkPoolWorker-6] update_subscribed create callback
[2024-03-17 17:26:06,563: WARNING/ForkPoolWorker-6] PLhIIJO_u92t5h-k0zjK55jCeT-AjuUBfi: get metadata from es
[2024-03-17 17:26:06,571: WARNING/ForkPoolWorker-6] PLhIIJO_u92t5h-k0zjK55jCeT-AjuUBfi: get metadata from youtube
[2024-03-17 17:26:07,273: WARNING/ForkPoolWorker-6] WARNING: [youtube:tab] Incomplete data received. Retrying (1/3)...
[2024-03-17 17:26:07,403: WARNING/ForkPoolWorker-6] WARNING: [youtube:tab] Incomplete data received. Retrying (2/3)...
[2024-03-17 17:26:07,512: WARNING/ForkPoolWorker-6] WARNING: [youtube:tab] Incomplete data received. Retrying (3/3)...
[2024-03-17 17:26:07,626: WARNING/ForkPoolWorker-6] WARNING: [youtube:tab] Incomplete data received. Giving up after 3 retries
[2024-03-17 17:26:07,708: WARNING/ForkPoolWorker-6] ab98fc35-31f5-4070-93a6-9054022559a2 success callback
[2024-03-17 17:26:07,712: INFO/ForkPoolWorker-6] Task update_subscribed[ab98fc35-31f5-4070-93a6-9054022559a2] succeeded in 1.3211056040599942s: None
[2024-03-17 17:26:07,712: WARNING/ForkPoolWorker-6] ab98fc35-31f5-4070-93a6-9054022559a2 return callback
```

### Anything else?

**Known workaround**

A workaround that I discovered is as below:
1. Go to the `/playlist` page. Click on the playlist in question.
2. Click on **Reindex** button. Waiting a few seconds.
3. Go to the `/downloads` page. Click on **Rescan subscriptions** button. We'll see new videos added into the download queue.

**Comment by lamusmaser** [comment]
We already have a PR for this in #670 - this should be fixed once that is merged.

**Comment by lamusmaser** [comment]
After a closer review of the PR, it does not actually resolve this issue. This only resolves a smaller action that is similar.

**Comment by bbilly1** [comment]
v0.4.8 brings various playlist handling fixes, and should now correctly trigger refreshs. please update.

**Comment by AdamGaffney96** [comment]
> v0.4.8 brings various playlist handling fixes, and should now correctly trigger refreshs. please update.

I have successfully updated to v0.4.8, however it still doesn't seem to work for me. I have a playlist I know for a fact is missing videos on my archive, however upon refresh these don't seem to get pulled through. Below are the logs when I refresh, I'm not sure if the missing stream and shorts tabs are causing this error but they always flag up when I rescan.

```
[2024-06-04 14:24:47,389: INFO/MainProcess] Task update_subscribed[f5cbfbbe-a01b-4de3-9b19-1f836b804201] received
04/06/2024
14:24:47
[2024-06-04 14:24:47,390: WARNING/ForkPoolWorker-4] update_subscribed create callback
04/06/2024
14:24:47
[2024-06-04 14:24:47,428: WARNING/ForkPoolWorker-4] UCaN8DZdc8EHo5y1LsQWMiig: find missing videos.
04/06/2024
14:24:48
[2024-06-04 14:24:48,937: WARNING/ForkPoolWorker-4] ERROR: [youtube:tab] UCaN8DZdc8EHo5y1LsQWMiig: This channel does not have a streams tab
04/06/2024
14:24:49
[2024-06-04 14:24:49,386: WARNING/ForkPoolWorker-4] ERROR: [youtube:tab] UCaN8DZdc8EHo5y1LsQWMiig: This channel does not have a shorts tab
04/06/2024
14:24:50
[2024-06-04 14:24:49,438: WARNING/ForkPoolWorker-4] UC6KvUTHUfCUF05gfnFTBHjA: find missing videos.
04/06/2024
14:24:50
[2024-06-04 14:24:50,906: WARNING/ForkPoolWorker-4] UC2aHABxfdl50OFV2VeXhv-w: find missing videos.
04/06/2024
14:24:52
[2024-06-04 14:24:52,310: WARNING/ForkPoolWorker-4] ERROR: [youtube:tab] UC2aHABxfdl50OFV2VeXhv-w: This channel does not have a streams tab
04/06/2024
14:24:53
[2024-06-04 14:24:53,065: WARNING/ForkPoolWorker-4] UCNvsIonJdJ5E4EXMa65VYpA: find missing videos.
04/06/2024
14:24:54
[2024-06-04 14:24:54,227: WARNING/ForkPoolWorker-4] ERROR: [youtube:tab] UCNvsIonJdJ5E4EXMa65VYpA: This channel does not have a streams tab
04/06/2024
14:24:54
[2024-06-04 14:24:54,727: WARNING/ForkPoolWorker-4] ERROR: [youtube:tab] UCNvsIonJdJ5E4EXMa65VYpA: This channel does not have a shorts tab
04/06/2024
14:24:54
[2024-06-04 14:24:54,768: WARNING/ForkPoolWorker-4] UCyNtlmLB73-7gtlBz00XOQQ: find missing videos.
04/06/2024
14:24:56
[2024-06-04 14:24:56,578: WARNING/ForkPoolWorker-4] ERROR: [youtube:tab] UCyNtlmLB73-7gtlBz00XOQQ: This channel does not have a shorts tab
04/06/2024
14:24:56
[2024-06-04 14:24:56,626: WARNING/ForkPoolWorker-4] UClt01z1wHHT7c5lKcU8pxRQ: find missing videos.
04/06/2024
14:24:58
[2024-06-04 14:24:58,080: WARNING/ForkPoolWorker-4] ERROR: [youtube:tab] UClt01z1wHHT7c5lKcU8pxRQ: This channel does not have a streams tab
04/06/2024
14:24:58
[2024-06-04 14:24:58,673: WARNING/ForkPoolWorker-4] ERROR: [youtube:tab] UClt01z1wHHT7c5lKcU8pxRQ: This channel does not have a shorts tab
04/06/2024
14:24:58
[2024-06-04 14:24:58,715: WARNING/ForkPoolWorker-4] UC5fdssPqmmGhkhsJi4VcckA: find missing videos.
04/06/2024
14:25:00
[2024-06-04 14:25:00,760: WARNING/ForkPoolWorker-4] ERROR: [youtube:tab] UC5fdssPqmmGhkhsJi4VcckA: This channel does not have a shorts tab
04/06/2024
14:25:00
[2024-06-04 14:25:00,807: WARNING/ForkPoolWorker-4] UC7-E5xhZBZdW-8d7V80mzfg: find missing videos.
04/06/2024
14:25:02
[2024-06-04 14:25:02,614: WARNING/ForkPoolWorker-4] ERROR: [youtube:tab] UC7-E5xhZBZdW-8d7V80mzfg: This channel does not have a shorts tab
04/06/2024
14:25:02
[2024-06-04 14:25:02,659: WARNING/ForkPoolWorker-4] UC5UQPZe-8v4_UP1uxi4Mv6A: find missing videos.
04/06/2024
14:25:04
[2024-06-04 14:25:04,747: WARNING/ForkPoolWorker-4] UCZB6V9fUov0Mx_us3MWWILg: find missing videos.
04/06/2024
14:25:06
[2024-06-04 14:25:06,157: WARNING/ForkPoolWorker-4] ERROR: [youtube:tab] UCZB6V9fUov0Mx_us3MWWILg: This channel does not have a streams tab
04/06/2024
14:25:06
[2024-06-04 14:25:06,635: WARNING/ForkPoolWorker-4] ERROR: [youtube:tab] UCZB6V9fUov0Mx_us3MWWILg: This channel does not have a shorts tab
04/06/2024
14:25:06
[2024-06-04 14:25:06,680: WARNING/ForkPoolWorker-4] UC2PA-AKmVpU6NKCGtZq_rKQ: find missing videos.
04/06/2024
14:25:08
[2024-06-04 14:25:08,680: WARNING/ForkPoolWorker-4] UCK-GxvzttTnNhq3JPYpXhqg: find missing videos.
04/06/2024
14:25:10
[2024-06-04 14:25:10,064: WARNING/ForkPoolWorker-4] ERROR: [youtube:tab] UCK-GxvzttTnNhq3JPYpXhqg: This channel does not have a streams tab
04/06/2024
14:25:10
[2024-06-04 14:25:10,628: WARNING/ForkPoolWorker-4] UCJ6o36XL0CpYb6U5dNBiXHQ: find missing videos.
04/06/2024
14:25:12
[2024-06-04 14:25:12,108: WARNING/ForkPoolWorker-4] ERROR: [youtube:tab] UCJ6o36XL0CpYb6U5dNBiXHQ: This channel does not have a streams tab
04/06/2024
14:25:12
[2024-06-04 14:25:12,593: WARNING/ForkPoolWorker-4] ERROR: [youtube:tab] UCJ6o36XL0CpYb6U5dNBiXHQ: This channel does not have a shorts tab
04/06/2024
14:25:12
[2024-06-04 14:25:12,658: WARNING/ForkPoolWorker-4] f5cbfbbe-a01b-4de3-9b19-1f836b804201 success callback
04/06/2024
14:25:12
[2024-06-04 14:25:12,662: INFO/ForkPoolWorker-4] Task update_subscribed[f5cbfbbe-a01b-4de3-9b19-1f836b804201] succeeded in 25.268323300988413s: None
04/06/2024
14:25:12
[2024-06-04 14:25:12,662: WARNING/ForkPoolWorker-4] f5cbfbbe-a01b-4de3-9b19-1f836b804201 return callback
```
Putting it in here as it seems related to the previous issue, but happy to add a new issue or provide extra info if necessary!

Thanks,

**Comment by Cyberes** [comment]
Copying my comment from https://github.com/tubearchivist/tubearchivist/issues/634#issuecomment-2263887553

>I'm having the same issue. Seems to be a regression since playlist subscriptions used to work fine. After upgrading the problem remains.
I tried reindexing the playlist but it isn't adding anything from it to the download queue.
Version 0.4.9
